mlos_vizpanel/frontend.py

```diff
@@ -0,0 +1,1134 @@
+    plt.ylabel("Result Score")
+    plt.title(f"{metric} Percentiles by Configuration")
+    plt.xticks(rotation=45)
+    plt.tight_layout()
+    st.pyplot(plt)
+
+# Function to plot whisker plots for configurations within a specific experiment
+
+def plot_whisker_plots(df, target_col, n=5):
+    """
+    Plots whisker plots for the top N and bottom N configurations with respect to a target column.
+
+    Parameters:
+    df (pd.DataFrame): The DataFrame containing the data to plot.
+    target_col (str): The name of the target column to plot on the y-axis.
+    n (int): The number of top and bottom configurations to plot.
+    """
+    if 'tunable_config_id' not in df.columns or target_col not in df.columns:
+        st.error(f"'tunable_config_id' or '{target_col}' column not found in DataFrame.")
+        return
+
+    # Ensure the target column is numeric and drop NaNs
+    df[target_col] = pd.to_numeric(df[target_col], errors='coerce')
+    df = df.dropna(subset=[target_col])
+
+    # Calculate the average of the target column for each configuration
+    config_avg = df.groupby('tunable_config_id')[target_col].mean().reset_index()
+
+    # Filter out configurations that do not have any result values
+    config_avg = config_avg.dropna(subset=[target_col])
+
+    # Select top N configurations
+    top_n_configs = config_avg.nlargest(n, target_col)['tunable_config_id']
+
+    # Filter the DataFrame to include only the top N configurations
+    top_configs = df[df['tunable_config_id'].isin(top_n_configs)]
+
+    # Sort the top configurations by the target column
+    top_configs = top_configs.sort_values(by=target_col, ascending=False)
+
+    # Plot whisker plots for the top N configurations
+    fig_top = px.box(top_configs, x='tunable_config_id', y=target_col,
+                     title=f"Whisker Plot for Top {n} Configurations by {target_col}",
+                     labels={'tunable_config_id': 'Configuration ID', target_col: target_col})
+    st.plotly_chart(fig_top, use_container_width=True)
+
+    # Select bottom N configurations
+    bottom_n_configs = config_avg.nsmallest(n, target_col)['tunable_config_id']
+
+    # Filter the DataFrame to include only the bottom N configurations
+    bottom_configs = df[df['tunable_config_id'].isin(bottom_n_configs)]
+
+    # Sort the bottom configurations by the target column
+    bottom_configs = bottom_configs.sort_values(by=target_col, ascending=True)
+
+    # Plot whisker plots for the bottom N configurations
+    fig_bottom = px.box(bottom_configs, x='tunable_config_id', y=target_col,
+                        title=f"Whisker Plot for Bottom {n} Configurations by {target_col}",
+                        labels={'tunable_config_id': 'Configuration ID', target_col: target_col})
+    st.plotly_chart(fig_bottom, use_container_width=True)
+
+
+# Function to plot correlation between parameter changes and latency
+def plot_param_latency_correlation(experiment_id, storage, metric):
```

Comment:
Think most of these `plot_*` methods could probably be split out to a separate set of modules for readability.

Reply:
A lot of them aren’t used, heads up

Reply:
Remove or comment them out for now then?